WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -50,6 +50,30 @@ def index
     User.find_by_id(params[:id] || current_user.id)
   end
 
+  def enable_2fa
+    # NOTE: current_user is not nil as authenticate_user! is called first
+    params[:section] = "2fa"
+    current_user.otp_required_for_login = true
+    current_user.otp_secret = User.generate_otp_secret
+    current_user.save!
+    if current_user.otp_required_for_login
```

Comment:
Note to self: this condition is ridiculous given the statement a couple line above.
I'll rework this so that it works as intended.